tutorials/grails3-send-push-notification-using-firebase-cloud-messaging

```diff
@@ -0,0 +1,196 @@
+   ```
+   import grails.plugins.rest.client.RestBuilder
+   ```
+3. Create a method to FcmController, so it will looks like this.
+   ```
+    package grails.fcm
+
+    import grails.plugins.rest.client.RestBuilder
+
+    class FcmController {
+
+    def index() { }
+
+    def sendPushNotification() {
+     def regid = params.regid
+     def title = params.title
+     def body = params.body
+
+     def rest = new RestBuilder(connectTimeout:1000, readTimeout:20000)
+     def resp = rest.post("https://fcm.googleapis.com/fcm/send") {
+       header 'Content-Type', 'application/json'
+       header 'Authorization', 'key=AIza*****'
+       json {
+         notification = {
+           title = title
+           body = body
+           sound = "default"
+           click_action = "FCM_PLUGIN_ACTIVITY"
+           icon = "fcm_push_icon"
+         }
+         to = regid
+       }
+     }
+
+     flash.message = "Notification sent"
+     redirect action: "index"
+    }
+   }
+   ```
+    Your previous written FCM legacy server key write on this line include with 'key='.
+   ```
+   header 'Authorization', 'key=AIza*****'
+   ```
+
+4. Create a view file "grails-app/views/fcm/index.gsp" and make this file like this.
+   ```
+    <!doctype html>
+    <html>
+    <head>
+     <meta name="layout" content="main"/>
+     <title>FCM Sender</title>
+
+     <asset:link rel="icon" href="favicon.ico" type="image/x-ico" />
+    </head>
+    <body>
+     <div id="content" role="main">
+         <section class="row colset-2-its">
+             <h1>Send Push Notification</h1>
+             <g:if test="${flash.message}">
+               <div class="message" role="alert">
+                 ${flash.message}
+               </div>
+             </g:if>
+             <g:form action="sendPushNotification">
+               <fieldset>
+                 <div class="fieldcontain">
+                   <label for="regid">FCM Registration ID</label>
+                   <g:textField name="regid" />
+                 </div>
+                 <div class="fieldcontain">
+                   <label for="title">Title</label>
+                   <g:textField name="title" />
+                 </div>
+                 <div class="fieldcontain">
+                   <label for="body">Message Body</label>
+                   <g:textField name="body" />
+                 </div>
+               </fieldset>
+               <fieldset>
+                 <div class="buttons">
+                   <g:submitButton class="save" name="submit" value="Send" />
+                 </div>
+               </fieldset>
+             </g:form>
+         </section>
+     </div>
+    </body>
+   </html>
+   ```
+5. Run and test push notification from the browser.
+
+## Cleaning Up
+
+After you have finished this tutorial, you might be want to clean up your Firebase project because the free tier limited to 3 projects. Just do the following:
```

Comment:
remove "you might be want to "

change "because the free tier limited to 3 projects" to "because you can only have 3 project in the free tier"